numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -2459,46 +2458,49 @@ einsum_list_to_subscripts(PyObject *obj, char *subscripts, int subsize)
             subscripts[subindex++] = '.';
             ellipsis = 1;
         }
-        /* Subscript */
-        else if (PyInt_Check(item) || PyLong_Check(item)) {
-            long s = PyInt_AsLong(item);
-            npy_bool bad_input = 0;
+        else {
+            npy_intp s = PyArray_PyIntAsIntp(item);
+            /* Subscript */
+            if (!PyErr_Occurred()) {
```

Comment:
```suggestion
            if (!error_converting(s)) {
```
A bit of a micro optimization, but its good to use generally and the more places use it, the more likely we don't forget it... (These functions return -1 on failure, so checking that first skips the `PyErr_Occured()` check in most cases).

Anyway, only glanced over but Anirudh and Eric had a look, so planning on merging it later. (I will apply this change myself).

Reply:
It might be a bit nicer to move the error into this if rather than the opposite way and then keep the rest unindented (as it was). I also feel its a bit more standard pattern handle the error immediately.